# IDENTITY PARAMETERS
Node_ID: 4
Motor_Name: joint_4

# CONTROL PARAMETERS
Max_Angle: 180
Min_Angle: -180
Home_Pos: 0

# MOTOR PARAMETERS
Motor_Type: 10 # DC (1), EC_SINUS (10), EC_BLOCK (11)
Motor_Nominal_Current: 2000        # MILLIAMPS
Motor_Max_Current: 4000           # MILLIAMPS
Motor_Thermal_Time_Constant: 18.4
Motor_Number_Pole_Pairs: 8         # EC_MAX (1), EC_4POLE (2)

# SENSOR PARAMETERS
Sensor_Type: 1                     # NONE (0), INC_3CH (1), INC_2CH (2)
                                   # HALL (3), SSI_BINARY (4), SSI_GRAY (5)
Sensor_Inversion:   0              # Encoder Opposite Hall (1), Same (0)
Encoder_Resolution: 512            # INC ENCODER CPR
Encoder_Data_Rate: 0               # ABS ENCODER SSI DATA RATE
Encoder_Number_Multiturn_Bits: 0   # ABS ENCODER SSI MULTI TURN BITS
Encoder_Number_Single_Turn_Bits: 0 # ABS ENCODER SSI SINGLE TURN BITS

# SAFETY PARAMETERS
Max_Following_Error: 2000   # MAX POS FOLLOWING ERROR, ENC COUNTS
Max_Profile_Velocity: 3000  # MAX PROFILE VELOCITY
Max_Acceleration: 15000      # MAX ACCELERATION

# PID PARAMETERS
PositionPGain : 328  # P GAIN FOR POSITION CONTROL
PositionIGain : 765  # I GAIN FOR POSITION CONTROL
PositionDGain : 739  # D GAIN FOR POSITION CONTROL
VelocityPGain : 1956  # P GAIN FOR VELOCITY CONTROL
VelocityIGain : 219  # I GAIN FOR VELOCITY CONTROL
CurrentPGain  : 510  # P GAIN FOR CURRENT CONTROL
CurrentIGain  : 99  # I GAIN FOR CURRENT CONTROL

# MODE SEPECIFIC PARAMETERS
Position_Profile_Velocity: 3000     # VELOCITY FOR POS PROFILE MODE
Position_Profile_Acceleration: 8000  # ACCELERATION FOR POS PROFILE MODE
Position_Profile_Deceleration: 14000 # DECELERATION FOR POS PROFILE MODE
Velocity_Profile_Acceleration: 8000  # ACCELERATION FOR VEL PROFILE MODE
Velocity_Profile_Deceleration: 9000  # DECELERATION FOR VEL PROFILE MODE
